compiler/scopelang/scope_to_dcalc.ml

```diff
@@ -530,7 +530,14 @@ let translate_rule (ctx : ctx) (rule : Ast.rule)
               (Dcalc.Ast.Var.make ("_", Pos.get_position e), Pos.get_position e);
             Dcalc.Ast.scope_let_typ = (Dcalc.Ast.TLit TUnit, Pos.get_position e);
             Dcalc.Ast.scope_let_expr =
-              Bindlib.box_apply (fun new_e -> Pos.same_pos_as (Dcalc.Ast.EAssert new_e) e) new_e;
+              (* To ensure that we throw an error if the value is not defined, we add an check
+                 "ErrorOnEmpty" here. *)
+              Bindlib.box_apply
+                (fun new_e ->
+                  Pos.same_pos_as
+                    (Dcalc.Ast.EAssert (Dcalc.Ast.ErrorOnEmpty new_e, Pos.get_position e))
```

Comment:
Neat, thanks!

Reply:
For the context, this permit us to handle EAssert the same way we handle addition or multiplication inside Dcalc, while raising errors if the input of the assertion is empty.

We belive that when we assert a property on a value, the value should always be well defined.